samcli/lib/sync/sync_flow.py

```diff
@@ -93,7 +103,15 @@ def set_up(self) -> None:
         self._session = Session(profile_name=self._deploy_context.profile, region_name=self._deploy_context.region)
 
     def _boto_client(self, client_name: str):
-        return get_boto_client_provider_from_session_with_config(cast(Session, self._session))(client_name)
+        default_retry_config = get_default_retry_config()
+        if not default_retry_config:
+            LOG.debug("Creating boto client (%s) with user's retry config", client_name)
```

Comment:
If we use logging framework's formatting, then it might not happen if the current logging level is not printing this message. For this example, if highest log level is set to `INFO` then all `DEBUG` messages won't be processed. So using in this way would have small performance improvement.

Reply:
O interesting... I didn't know that but TIL :)